### PR TITLE
container: T4959: Add container registry authentication config for containers

### PR DESCRIPTION
**Pull request by Zen3515**
## Change Summary

Add container registry authentication config for containers

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4959

## Component(s) name

- containers

## Proposed changes

- Add `disable`, `username`, and, `password` to `container > registry` for authentication
- Add `default-registry` to container configuration to store default registry for using alongside registry config  

The way this works is by using the provided username and password to login with command `podman login` and using `podman logout --all` to remove all the credentials.

## How to test

```bash
$ set container registry example.com
$ commit
$ set container registry docker.io username alice
$ set container registry docker.io password alices_password
$ commit
$ exit
$ add container image SOME_LOCAL_OR_PRIVATE_REPO
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/container.xml.in [hunk 1]
```diff
@@ -368,9 +368,33 @@
           </leafNode>
         </children>
       </tagNode>
-      <leafNode name="registry">
+      <tagNode name="registry">
```
Comment (sever-sever) [inline interface-definitions/container.xml.in:371]: The default value for the registry should be `docker.io quay.io` if not configured explicitly.
But if the `registry` is configured, It shouldn't use the default values but only configured registry entries.

Comment (Zen3515) [reply interface-definitions/container.xml.in:371]: I see, where's the place that I should store `docker.io quay.io`, is it inside .py file or j2 file?

Comment (sever-sever) [reply interface-definitions/container.xml.in:371]: @c-po as I understand it possible to use default XML values with tagNode? Or  is it better to use Python for this case?

Comment (c-po) [reply interface-definitions/container.xml.in:371]: We must always use `<defaultValue>` to have the defaults in one unique place. I think it is currently not possible and needs a schema extension.

Comment (c-po) [reply interface-definitions/container.xml.in:371]: Will be added in https://vyos.dev/T5079

**Comment by c-po** [comment]
Implemented via:

* b4af532dd53 ("schema: [T5079](https://vyos.dev/T5079): extension to support defaultValues on tagNodes")
* fe82d86d3e8 ("container: [T4959](https://vyos.dev/T4959): add registry authentication option")

**Pull request closed without merging**